### PR TITLE
127 tactics set timing over counting

### LINKED ISSUE #127: Tactics Set Timing Over-counting
Customer Submission:
Since yesterday I have an issue with the woodpecker training. The added time after exit the traing is multiple times more than spent in the training (about 4-5 times more...) so I have now in the 7th round much more time displayed than the rounds before.

### PR DESCRIPTION
**Pull request by KeeghanM**
Fixes issue #127 

This pull request updates the tactics trainer component to improve how puzzle timing is tracked and reported, and cleans up some code related to time tracking on exit. The most important changes are grouped below.

**Puzzle timing and state management:**

* Changed `startTime` to use a setter (`setStartTime`) so it can be updated after each puzzle, allowing for accurate tracking of time spent per puzzle.
* After a correct answer, the time taken for the puzzle is now calculated using a fresh timestamp (`newTime`), and `startTime` is reset for the next puzzle. This ensures each puzzle's time is measured independently.

**Code cleanup:**

* Removed the call to `increaseTimeTaken.mutate` from the `exit` function, so time is only tracked per puzzle and not redundantly on exit.

**Minor UI/comment updates:**

* Improved comment formatting for the loading spinner conditional and fixed its indentation for clarity. [[1]](diffhunk://#diff-5e258e2c7b983c33001cb2cca94781092999d9355c9c88096823f1cc829bd623L375-R374) [[2]](diffhunk://#diff-5e258e2c7b983c33001cb2cca94781092999d9355c9c88096823f1cc829bd623L384-R384)

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved accuracy of time tracking for tactics puzzles by consistently capturing and resetting timestamps.
  - Prevented incorrect time records when exiting a puzzle by removing the previous exit-time mutation.
  - Loading indicator behavior refined so it reflects puzzle fetching and round creation more reliably.

### CODE REVIEW

File: src/app/components/training/tactics/TacticsTrainer.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default function TacticsTrainer(props: {
   // Setup state for the game and training
   const [readyForInput, setReadyForInput] = useState(false)
   const [puzzleFinished, setPuzzleFinished] = useState(false)
-  const [startTime] = useState(Date.now())
+  const [startTime, setStartTime] = useState(Date.now())
```
[the remark this answers is not in the document]
Comment (KeeghanM) [reply src/app/components/training/tactics/TacticsTrainer.tsx:82]: Is this a good way to do this?
```
 // Increase timer whenever puzzle is finished
  useEffect(() => {
    if (puzzleFinished) {
      increaseTimeTaken.mutate({
        roundId: currentRound.id,
        timeTaken: (Date.now() - startTime) / 1000,
        setId: props.set.id,
      })
    } else {
      setStartTime(Date.now())
    }
  }, [puzzleFinished])
  ```
  
  That way, whether it was right or wrong - we'll log the time. And we'll reset the start timer whenever the puzzle starts (aka, is not finished)
